src/systems/acoustic_comms/AcousticComms.cc

```diff
@@ -175,8 +190,41 @@ void AcousticComms::Step(
         {
           if (distanceCoveredByMessage >= distanceToTransmitter)
           {
+            // This message has effectively reached the destination.
+            bool receivedSuccessfully = false;
+
+            // Check for time collision
+            if (this->dataPtr->lastMsgReceivedTime.count(
+                  msg->dst_address()) == 0)
+            {
+              // This is the first message received by this address.
+              receivedSuccessfully = true;
+            }
+            else
+            {
+              // A previous msg was already received at this address.
+              std::chrono::duration<double> timeGap = currTimestamp -
+                this->dataPtr->lastMsgReceivedTime[msg->dst_address()];
+
+              auto dropInterval = std::chrono::duration<double>(
```

Comment:
Knit: From a logical stand point wouldn't the dropped packet also clog the reciever for a fixed amount of time.

Reply:
Done https://github.com/gazebosim/gz-sim/pull/1755/commits/25b975ae3b6d7c0f48ad3a428bf1a951179b01dc